dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,245 @@
+    except Exception as e:
+        _LOG.error("Error saving label inventory manifest file: %s", str(e))
+        raise e
+
+
+def _parse() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    hparser.add_verbosity_arg(parser)
+    parser.add_argument(
+        "--input_file",
+        required=True,
+        help="Path to label inventory manifest file",
+    )
+    parser.add_argument(
+        "--owner",
+        required=True,
+        help="GitHub repository owner/organization",
+    )
+    parser.add_argument("--repo", required=True, help="GitHub repository name")
+    parser.add_argument(
+        "--token_env_var",
+        required=True,
+        help="Name of the environment variable containing the GitHub token",
+    )
+    parser.add_argument(
+        "--prune",
+        action="store_true",
+        help="Delete labels that exist in the repo but not in the label inventory manifest file",
+    )
+    parser.add_argument(
+        "--dry_run",
+        action="store_true",
+        help="Print out the actions that would be taken without executing them",
+    )
+    parser.add_argument(
+        "--backup",
+        action="store_true",
+        help="Backup current labels to a label inventory manifest file",
+    )
+    return parser
+
+
+def _main(parser: argparse.ArgumentParser) -> None:
+    args = parser.parse_args()
+    hdbg.init_logger(verbosity=args.log_level, use_exec_path=True)
+    # Load labels from label inventory manifest file.
+    labels = _load_labels(args.input_file)
+    labels_map = {label.name: label for label in labels}
+    token = os.environ.get(args.token_env_var, None)
+    hdbg.dassert(token)
+    # Initialize GH client.
+    client = github.Github(token)
+    repo = client.get_repo(f"{args.owner}/{args.repo}")
+    # Get current labels from the repo.
+    current_labels = repo.get_labels()
+    current_labels_map = {label.name: label for label in current_labels}
+    # Execute code if not in dry run mode.
+    execute = not args.dry_run
+    # Save the labels if backup is enabled.
+    if args.backup:
+        root_dir = hgit.get_client_root(False)
+        dst_dir = f"{root_dir}/dev_scripts_helpers/github/labels/backup"
+        file_name = f"tmp.labels.{args.owner}.{args.repo}.yaml"
+        file_path = f"{dst_dir}/{file_name}"
+        if execute:
+            _save_labels(current_labels, file_path)
+        _LOG.info("Labels backed up to %s", file_path)
+    # Confirm label synchronization.
+    hsystem.query_yes_no(
+        "Are you sure you want to synchronize labels?", abort_on_no=True
+    )
+    # Delete labels if pruning is enabled.
+    if args.prune:
+        for current_label in current_labels:
+            if current_label.name not in labels_map:
+                if execute:
+                    current_label.delete()
+                _LOG.info("Label deleted: %s", current_label.name)
```

Comment:
Should go in the if-then and you can do

else:
   _LOG.info("The label %s will be deleted without --dry-run"